test/integration/api/applications_test.rb

```diff
@@ -92,6 +92,16 @@ class ApplicationsTest < ActionDispatch::IntegrationTest
     assert_match(/[A-Za-z0-9]\w+/, body.fetch("oauth_secret"))
   end
 
+  test "#create with no permissions is successful" do
+    post_req(endpoint, params: create_params.merge("permissions" => []))
+    assert_equal 200, response.status
+    body = JSON.parse(response.body)
+    assert_equal 43, body.fetch("oauth_id").length
+    assert_match(/[A-Za-z0-9]\w+/, body.fetch("oauth_id"))
```

Comment:
Just to double-check the regex here as it seems oddly specific: is it the case that our OAuth client IDs and secrets can contain `_` but not start with `_`?

Reply:
Thanks, I've pushed up another commit to correct the regex.

Reply:
I think you're now just checking that it contains at least one alpha (or `-` or `_`) character. Did you drop the `+` intentionally? Also I just realised these regexes are unanchored so if you wanted to check that the whole string consists of only those characters you'd need something like `/^[A-Za-z0-9_-]+$/`

Reply:
I merged too soon, sorry - I've put up a separate PR for this https://github.com/alphagov/signon/pull/1643